.github/workflows/build-soroban-dev.yml

```diff
@@ -0,0 +1,43 @@
+name: Soroban-Dev
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+
+# Prevent more than one build of this workflow for a branch to be running at the
+# same time, and if multiple are queued, only run the latest, cancelling any
+# already running build. The exception being any protected branch, such as
+# master, where a build for every commit will run.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_protected == 'true' && github.sha || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  complete:
+    if: always()
+    needs: [build]
+    runs-on: ubuntu-latest
+    steps:
+    - if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
+      run: exit 1
+
+  build:
+    uses: ./.github/workflows/build.yml
+    secrets:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+    with:
+      # The `:soroban-dev` tag points to a build containing unreleased versions
+      # of software that have been informally released to the futurenet network.
+      tag: soroban-dev
+      core_ref: c0ad35aa19297e112d71fcc5755458495f99a237
+      go_ref: soroban-v0.0.4
+      soroban_tools_ref: v0.4.0
```

Comment:
it would be better if we can derived the dependencies here : 
the soroban-tools contains a reference to the go repo, and the go repo contains a reference to the core.
when it comes to building it here, we should "unfold" all these.

Reply:
To a degree. I think there's an argument for simplicity over clever if we don't need it. These also might not always be derivable. Core typically updates before Horizon for example. We don't want to wait to update Core until Horizon is ready.